sklearn/utils/estimator_checks.py

```diff
@@ -1338,3 +1343,44 @@ def fit(self, X, y):
 
     assert_true(all(item in deep_params.items() for item in
                     shallow_params.items()))
+
+
+@ignore_warnings
+def check_estimator_fit_reset(name, Estimator):
+    X1, y1 = make_blobs(n_samples=50, n_features=2, center_box=(-200, -150),
+                        centers=2, random_state=0)
+    X2, y2 = make_blobs(n_samples=50, n_features=2, center_box=(200, 150),
+                        centers=2, random_state=1)
+    X3, y3 = make_blobs(n_samples=50, n_features=2, center_box=(-200, 150),
+                        centers=3, random_state=2)
+    X4, y4 = make_blobs(n_samples=50, n_features=5, center_box=(-200, -150),
+                        centers=2, random_state=0)
+    X5, y5 = make_blobs(n_samples=50, n_features=5, center_box=(200, 150),
+                        centers=2, random_state=1)
+    X6, y6 = make_blobs(n_samples=50, n_features=5, center_box=(-200, 150),
+                        centers=3, random_state=2)
+
+    # Some estimators work only on non-negative inputs
+    if name in ('AdditiveChi2Sampler', 'SkewedChi2Sampler',
+                'NMF', 'MultinomialNB', 'ProjectedGradientNMF'):
+        X1, X2, X3, X4, X5, X6 = map(np.fabs, (X1, X2, X3, X4, X5, X6))
```

Comment:
It might not matter, but doesn't this break the blobiness of the data? I'd rather just subtract the featurewise minimum value.